_drafts/bazel-for-embedded.md

```diff
@@ -0,0 +1,389 @@
+
+The behavior and configuration of Bazel can be changed through command line arguments (also called flags). A file named `.bazelrc` can be used at the root of the workspace to collect all flags that should be added when invoking Bazel.
+
+## Platforms
+
+Platforms is a Bazel concept to deal with build variants. This concept addresses many of the challenges in embedded development, for example when dealing with different hardware, having multiple products with different features, handling different product versions, using different compilers, sharing software components, testing on different architectures, and many more...
+
+To understand how platforms are used with Bazel, there are a few fundamentals to understand first.
+
+A platform is a way for Bazel to model the environment of the build, it is a very abstract notion that can describe many things such as hardware, a CPU, a software feature, etc. There are 3 kinds of platforms:
+
+- `host`: the platform on which Bazel itself runs (your Linux machine for example).
+- `execution`: the platform on which the tools invoked by Bazel runs to generate build artifacts (your Linux machine or an Amazon server where the build is taking place with remote execution for example).
+- `target`: the platform on which the final output resides and executes (a Raspberry PI or a software version or just a feature such as allowing runtime exceptions for testing for example).
+
+In a cross-compilation context, the **host** and **execution** platforms are often the same. So for the following, we will focus on this scenario.
+
+> Note that Bazel will generate the host and execution platform by automatically detecting the type of machine it is running on.
+
+A platform is simply a collection of **constraints** (you can see a constraint as an identifier for now). To choose a compiler, Bazel will try to match the constraints defined by the compiler for the execution platform and the target platform, this step is called the **toolchain resolution**.
+
+For example, imagine we defined a platform called `esp32` that defines the constraint `@platforms//cpu:xtensa`. When Bazel is invoked with the argument `--platforms=esp32`, the execution platform is automatically generated by Bazel (it contains constraints that define your current machine, for example, `@platforms//cpu:x86_64` while the target platform is the one passed into an argument which defines the constraints `@platforms//cpu:xtensa`).
+
+With this pair of constraints, during the toolchain resolution process, Bazel will try to find the appropriate toolchain to compile your code.
+
+This process is transparent to the user and therefore, if multiple toolchains are available for various execution platforms (for example, one for Linux, one for Windows), the user can seamlessly compile the same code on different hosts running the same command without even thinking about it.
+
+## Toolchains
```

Comment:
Could you add a section before `## Toolchains` that does the following:

1. intro to executing `bazel` from the command line (just a few examples to set the context, and link to a full tutorial), explanation of a couple of the basic commands (`bazel run` eg)
2. step-by-step explained `.bazelrc` file (eg as in the below `cc_binary()` snippet) that builds a basic hello world c/cpp program, to provide context

Otherwise we're jumping right into toolchain management without setting some background on basic bazel project setup

Reply:
Yes make sense, I will add the section before "Platforms" even, so just after introducing "Bazel".